### PR TITLE
feat: capture stdout and store as output

### LINKED ISSUE #274: Example of how to use this and get output of one or more lines to `$GITHUB_OUTPUT`
I'm trying to look at the output from one of my multiline scripts in a follow up job or step. If a particular string is found, then the workflow set to fail.

Is there a way to use the relatively new methods for defining output parameters using `$GITHUB_OUTPUT`? ([docs](https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs))

If so, it would be great to see an example for `ssh-actions` including the `outputs` `id` and `needs` as in [this example](https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs#example-defining-outputs-for-a-job).

### PR DESCRIPTION
**Pull request by GammaGames**
Possibly fixes #274, #38

```yaml
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: SSH
        id: ssh
        uses: https://github.com/GammaGames/ssh-action@master
        with:
          host: [fill in]
          capture_stdout: 'true'
          script: |
            ls -l ~/test/
      - name: echo
        run: echo "${{ steps.ssh.outputs.stdout }}"
```

If successful, it has the following in `steps.ssh.outputs.stdout`:
```
======CMD======
ls -l ~/test/
======END======
out: total 0
out: -rw-rw-r-- 1 user group 0 Jan  2 14:24 file1
out: -rw-rw-r-- 1 user group 0 Jan  2 14:24 file2
==============================================
✅ Successfully executed commands to all host.
==============================================
```

On error it acts like it does now.

### CODE REVIEW

**Comment by GammaGames** [comment]
fixed conflicts

**Comment by hasechris** [comment]
@appleboy 
Could you take a look at this?

**Comment by Digital-Lemon** [comment]
I'd add a variable to enable (or disable) this behavior, as sometimes the outputs could be huge

**Comment by GammaGames** [comment]
Good point, it should *probably* also not capture by default because the output has all command output (without censoring, like the gh step interface has)

**Comment by uchar** [comment]
I think the script should handle masking secrets by itself , not  very good practice to depend on another step for doing it , this can cause security issues if some one forget it

**Comment by GammaGames** [comment]
I agree! Is there a way to get secret names at runtime? I was under the impression that github action output is filtered before displaying, but doesn't have any filtering on the server side

**Comment by mikesalvia** [comment]
@GammaGames or @appleboy is there any update on when this would be merged in?

**Comment by GammaGames** [comment]
@mikesalvia I don't know, but I should probably make it optional and disabled by default first. I started working on that a little today but have yet to test it: https://github.com/GammaGames/ssh-action/tree/capture_vars

**Comment by appleboy** [comment]
I will take it.

File: action.yml [hunk 1]
```diff
@@ -77,6 +77,13 @@ inputs:
     description: "pass all environment variable to shell script."
   request_pty:
     description: "Request a pseudo-terminal from the server."
+  capture_stdout:
+    description: "Capture the stdout of the commands."
+    default: "false"
+
+outputs:
+  stdout:
```
[the remark this answers is not in the document]
Comment (GammaGames) [reply action.yml:85]: Fixed, thanks! Couldn't figure out what the problem was

**Comment by appleboy** [comment]
@GammaGames Please help to resolve the conflicts.

**Comment by GammaGames** [comment]
Fixed, also changed the test to use `true` instead of `"true"` (though both work)

Why remove the `script_stop` feature?

**Comment by appleboy** [comment]
@GammaGames Thanks.

> Why remove the script_stop feature?

The script_stop function detects each line of instruction, but it cannot handle line breaks using the backslash (\). Therefore, I will write another method to solve this problem.

**Comment by banagale** [comment]
Thank you @GammaGames and all who assisted!

**Pull request merged**